src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotAlwaysSkipTokenValidationInDelegatesTests.cs

```diff
@@ -0,0 +1,602 @@
+                            .AddMetadataReferences(solution.GetProject(projectId).MetadataReferences)
+                            .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                        return sideProject.Solution.GetProject(projectId)
+                            .AddProjectReference(new ProjectReference(sideProject.Id))
+                            .Solution;
+                    }
+                }
+            }.RunAsync();
+        }
+
+        // Ideally we could detect this but we'll have to rely on CodeQL instead for more robust detection.
+        [Fact]
+        public async Task TestDelegatedMethodFromLocalFromDifferentAssemblyNoDiagnostic()
+        {
+            string source1 = @"
+using System;
+using System.Collections.Generic;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AcceptAllAudiencesNamespace
+{
+    public class AcceptAllAudiencesClass
+    {
+        public static bool AcceptAllAudiences2(
+            IEnumerable<string> audiences,
+            SecurityToken securityToken,
+            TokenValidationParameters validationParameters)
+        {
+            return true;
+        }
+    }
+}";
+
+            var source2 = @"
+using System;
+using System.Collections.Generic;
+using Microsoft.IdentityModel.Tokens;
+using AcceptAllAudiencesNamespace;
+
+class TestClass
+{
+    public bool AcceptAllAudiences(
+        IEnumerable<string> audiences,
+        SecurityToken securityToken,
+        TokenValidationParameters validationParameters)
+    {
+        return AcceptAllAudiencesClass.AcceptAllAudiences2(audiences, securityToken, validationParameters);
+    }
+
+    public void TestMethod()
+    {
+        TokenValidationParameters parameters = new TokenValidationParameters();
+        parameters.AudienceValidator = new AudienceValidator(AcceptAllAudiences);
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithWilson,
+                TestState =
+                {
+                    Sources = { source2 },
+                },
+                SolutionTransforms =
+                {
+                    (solution, projectId) =>
+                    {
+                        var sideProject = solution.AddProject("DependencyProject", "DependencyProject", LanguageNames.CSharp)
+                            .AddDocument("Dependency.cs", source1).Project
+                            .AddMetadataReferences(solution.GetProject(projectId).MetadataReferences)
+                            .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                        return sideProject.Solution.GetProject(projectId)
+                            .AddProjectReference(new ProjectReference(sideProject.Id))
+                            .Solution;
+                    }
+                }
```

Comment:
Would something like this work?

https://github.com/dotnet/roslyn-analyzers/blob/762b08948cdcc1d94352fba681296be7bf474dd7/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableCertificateValidationTests.cs#L197-L211

Reply:
It appears this syntax isn't yet supported in the 2.9.x branch @dotpaul recommended I target. Will leave as is which matches the implementation for the class you referenced in 2.9.x branch.